UI/src/pages/MaintenancePlanPage.js

```diff
@@ -1,48 +1,84 @@
 import React, { Component } from 'react';
-import TimePicker from 'material-ui/TimePicker';
-import DatePicker from 'material-ui/DatePicker';
+import DateTimePicker from 'material-ui-datetimepicker';
+import DatePickerDialog from 'material-ui/DatePicker/DatePickerDialog'
+import TimePickerDialog from 'material-ui/TimePicker/TimePickerDialog';
 import TextField from 'material-ui/TextField';
 import SelectField from 'material-ui/SelectField';
 import MenuItem from 'material-ui/MenuItem';
 import RaisedButton from 'material-ui/RaisedButton';
 import Header from '../components/Header';
 import BackButton from '../components/BackButton';
+import dbapi from '../apirequests/dbapi';
+import jwt from 'jsonwebtoken';
 
 //it contains maintenance plan form
 class MaintenancePlanPage extends Component {
     constructor(props) {
         super(props);
 
         this.state = {
-            name: '',
-            date: '',
-            time: '',
-            machine: '',
-            location: ''
+            user_id: null,
+            start_date_time: null,
+            equipment_id: -1,
+            location_id: -1,
+            equipment:[],
+            locations:[]
         }
     }
 
-
-    handleChangeTimePicker(event, time) {
-        this.setState({ time: time });
-    };
-
-    handleChangeName(event) {
-        this.setState({ name: event.target.value });
-    };
-
-    handleChangeDatePicker(event, date) {
-        this.setState({ date: date });
-    };
+    handleChangeDateTimePicker = (dateTime) => this.setState({ start_date_time: dateTime});
 
     handleChangeMachine(event, index, value) {
-        this.setState({ machine: value })
+        this.setState({ equipment_id: value })
     };
 
     handleChangeLocation(event, index, value) {
-        this.setState({ location: value })
+        this.setState({ location_id: value })
     };
 
+    handlePostRequest(){
+      this.setState({user_id: jwt.decode(localStorage.getItem('token')).user_id});
+      this.state.start_date_time.toJSON();
+      dbapi.post('/maintenance', {
+        params: {
+          user_id: this.state.user_id,
+          start_date_time: this.state.start_date_time,
+          equipment_id: this.state.equipment_id,
+          location_id: this.state.location_id
+        }
+      })
+        .then(function (response) {
+          console.log("posted");
+        })
+        .catch(function (response) {
+          console.log("Error: " + response);
+        });
+    }
+
+    componentWillMount() {
+  		this.updateData();
+  	}
+
+  	updateData() {
+  		let user_id = jwt.decode(localStorage.getItem('token')).user_id;
+      this.state.user_id = user_id;
```

Comment:
use this.setState instead of setting the setting the state directly. Also fix the tabbing